packages/app/cypress/e2e/cypress-in-cypress-component.cy.ts

```diff
@@ -145,4 +145,67 @@ describe('Cypress In Cypress CT', { viewportWidth: 1500, defaultCommandTimeout:
       )
     })
   })
+
+  it('shows automation disconnected warning', () => {
+    cy.visitApp()
+    cy.get('[data-cy="spec-item"]').first().click()
+    // Let runner stabilize
+    cy.get('[data-cy="reporter-panel"]').should('be.visible')
+
+    cy.withCtx((ctx) => {
+      ctx.coreData.servers.appSocketServer?.emit('automation:disconnected')
+    })
+
+    cy.contains('h3', 'The Cypress extension has disconnected')
+
+    cy.withCtx((ctx, { sinon }) => {
+      sinon.stub(ctx.actions.project, 'launchProject').resolves()
+    })
+
+    cy.contains('button', 'Reload the browser').click()
+
+    cy.withCtx((ctx) => {
+      expect(ctx.actions.project.launchProject).to.have.been.called
+    })
+  })
+
+  it('shows automation missing warning', () => {
+    let connectedCallback: any
+
+    cy.visitApp()
+
+    cy.window().then((win) => {
+      const originalEmit: Function = win.ws?.emit || function () {}
```

Comment:
Neat trick. Since `win.ws.emit` must be defined for this test to pass, should we just throw an error if it's not defined? If we ever introduce a race condition that makes `win.ws` not defined, it'll be a lot easier to track down an explicit error than the classic "cannot call emit of undefined".

Reply:
Addressed in e1ac0624350e26daac737eace6e2fecde8bdaa85